js/main.js

```diff
@@ -106,10 +113,28 @@ function switchView(view) {
     }
   }
   if (view === 'search-home') {
-    $movieJournalNav.setAttribute('class', 'movie-journal-anchor white font-roboto hidden');
+    $movieJournalNav.className = 'movie-journal-anchor white font-roboto hidden';
+    $colNav.className = 'col-nav justify-right';
+  }
+  if (view === 'search-results') {
+    $colNav.className = 'col-nav justify-right';
   }
   if (view === 'search-result-detailed') {
-    $movieJournalNav.setAttribute('class', 'movie-journal-anchor white font-roboto');
+    $movieJournalNav.className = 'movie-journal-anchor white font-roboto';
+    $colNav.className = 'col-nav justify-right';
+  }
+  if (view === 'watchlist') {
+    $movieJournalNav.className = 'movie-journal-anchor white font-roboto';
+    $colNav.className = 'col-nav justify-right hidden';
+    for (var n = 0; n < data.savedCards.length; n++) {
+      var watchlistDomTree = renderWatchlist(data.savedCards[n]);
+      $ulWatchlist.appendChild(watchlistDomTree);
+    }
+  }
```

Comment:
This seems to have a lot of repetition in it. What work is this doing exactly? Is this handling some sort of special view?